syntax/pb_test.go

```diff
@@ -0,0 +1,61 @@
+package syntax
+
+import (
+	"testing"
+
+	"github.com/arr-ai/arrai/rel"
+	"github.com/arr-ai/wbnf/parser"
+	"github.com/stretchr/testify/assert"
+)
+
+const sharedCode = `let sysl = //encoding.proto.decode(//encoding.proto.proto, ` +
+	`//os.file('../translate/pb/test/sysl.pb'));` +
+	`let decodeSyslPb = //encoding.proto.decode(sysl);` +
+	`let shop = decodeSyslPb('Module', //os.file("../translate/pb/test/petshop.pb"));`
+
+func TestTransformProtobufToTupleFlow(t *testing.T) {
+	t.Parallel()
+
+	code := sharedCode + `shop`
+	pc := ParseContext{SourceDir: ".."}
+	ast, err := pc.Parse(parser.NewScanner(code))
+	assert.NoError(t, err)
+
+	codeExpr := pc.CompileExpr(ast)
+	val, err := codeExpr.Eval(rel.EmptyScope)
+
+	assert.NoError(t, err)
+	shop, _ := val.(rel.Tuple)
+	apps, _ := shop.Get("apps")
+
+	assert.NotNil(t, apps)
+}
+
+func TestTransformProtobufToTupleMapList(t *testing.T) {
+	t.Parallel()
+
+	code := sharedCode + `shop.apps('PetShopApi').name.part.@`
+	AssertCodesEvalToSameValue(t, `0`, code)
+
+	code = sharedCode + `shop.apps('PetShopApi').name.part.@item.@item`
+	AssertCodesEvalToSameValue(t, `'PetShopApi'`, code)
+
+	code = sharedCode + `shop.apps('PetShopApi').attrs('package').s`
+	AssertCodesEvalToSameValue(t, "'io.sysl.demo.petshop.api'", code)
+
+	code = sharedCode + `shop.apps('PetShopApi').endpoints('GET /petshop').attrs('patterns').a.elt(0).@item.s`
+	AssertCodesEvalToSameValue(t, "'rest'", code)
+
+	code = sharedCode + `shop.apps('PetShopApi').endpoints('GET /petshop').attrs('patterns').a.elt.@item.@item.s`
+	AssertCodesEvalToSameValue(t, "'rest'", code)
```

Comment:
A flat array can't get element index by `@`, for example: `petshop.apps('PetShopApi').endpoints('GET /petshop').attrs('patterns').a.elt.@` will print indedx `0`.

Reply:
Sure, but `a.elt.@item.s` should work, right? Since there's only one element.

Reply:
Now code transforms list to a tuple array, so it is a tuple array still even it has only one element. That's why it can use `a.elt.@item.@item.s`, and actually it can run `a.elt(0).@item.s` to access too. So I think `a.elt(0).@item.s` can work for you too.

Reply:
Right, but what I'm saying is there should only be one level of array in this case.

- `a` is an `Array` message, so it's a tuple with a `part` field.
- `part` is a `repeated string` field so it should be an array, not an array of arrays.

Reply:
Sorry, yes, I just read the proto wrong: `a` is an `Array` with repeated `Attribute`s, each of which is a `repeated string`, so it is indeed an array of arrays, not a flat array. Carry on!